client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/tcp/TcpTransportFactory.java

```diff
@@ -291,6 +291,12 @@ public Transport getTransport(Object key, Set<SocketAddress> failedServers, byte
 
    @Override
    public void releaseTransport(Transport transport) {
+      if (transport.isBusy()) {
```

Comment:
Because RetryOnFailureOperation would attempt to release the transport at the end of the operation, but in the case of the streaming ops this doesn't happen until the stream is closed by the user.